chore: remove antd from correlation table

daibhin opened the pull request:
## Problem

Towards https://github.com/PostHog/posthog/issues/13624

## Changes

- Ported both tables over
- Remove some styles in favour of Tailwind classes

Plenty could be shared here but I just wanted to get the minimum done for antd removal.

## How did you test this code?

- Added a story

|Before|After|
|----|----|
| <img width="673" alt="Screenshot 2024-03-11 at 17 53 50" src="https://github.com/PostHog/posthog/assets/6685876/ec0717c6-dd63-41ba-bc58-ddf282247bf0"> | <img width="916" alt="Screenshot 2024-03-12 at 10 40 19" src="https://github.com/PostHog/posthog/assets/6685876/10a05c6e-2e1c-4a96-b1e1-f68549a89c15"> |

<img width="918" alt="Screenshot 2024-03-12 at 10 39 48" src="https://github.com/PostHog/posthog/assets/6685876/5e54ef49-b163-49cb-9971-796359692c4c">
<img width="921" alt="Screenshot 2024-03-12 at 10 39 38" src="https://github.com/PostHog/posthog/assets/6685876/03211bfc-f944-4bce-85ba-ce206273feb5">
<img width="1191" alt="Screenshot 2024-03-12 at 09 35 45" src="https://github.com/PostHog/posthog/assets/6685876/b48d6eb2-dc2a-4af5-be04-6f5ec5a9a1a1">
`(Empty string)` here is just because I didn't go to the effort of mocking the correlation endpoint twice for the nested properties and the parsing then doesn't work. Should be fine in production

Commit:
chore: remove antd from correlation table

Commit:
do not allow antd table components

daibhin commented on frontend/src/mocks/fixtures/api/projects/team_id/insights/funnelCorrelation.json:
```diff
@@ -0,0 +1,48 @@
+{
```
Random data I came up with based on an example response from production

daibhin commented on frontend/src/scenes/insights/views/Funnels/CorrelationActionsCell.tsx:
```diff
@@ -82,8 +82,16 @@ const CorrelationActionsCellComponent = ({ buttons }: CorrelationActionsCellComp
+                    key={index}
+                    fullWidth
+                    onClick={(e) => {
+                        setPopoverOpen(false)
```
Improvement over existing. Popover stayed open when you opened the modal 🤮

thmsobrmlr reviewed (changes requested):
Thanks for cleaning this up!

You already saw the other PR that's not causing the merge conflicts here. We can revert that or remove the antd style adaptations from there once antd is fully gone.

Found a couple of issues here though:
1. The pagination for correlated events doesn't work. My local insight should have 4 pages with 5 items (see the 20 items before I click), but only 2 pages are displayed. Clicking on the second page causes this:
    
    ![2024-03-12 12 13 27](https://github.com/PostHog/posthog/assets/1851359/0dcc09b2-625f-4f64-b8c4-66ac199b4f71)

2. Uncollapsing another event should not trigger a reload of the previously opened ones:

    ![2024-03-12 12 13 56](https://github.com/PostHog/posthog/assets/1851359/70ced5b3-683b-4ef0-b478-c5ab1326e165)

3. (minor): The white styling here looks a bit off, since the rest is tan:

    <img width="1598" alt="Screenshot 2024-03-12 at 12 14 15" src="https://github.com/PostHog/posthog/assets/1851359/c842dbeb-da31-4da4-8f8b-92d4ac246986">

    Ideally we have a similar styling to before (tan header, white content) I guess. Not a blocker and I know that's subjective as well.

Commit:
Merge branch 'master' into dn-chore/funnel-correlation-table

daibhin commented:
@thmsobrmlr thanks for taking a look.

I reverted to using your empty state. It seems like Product Analytics use the `Empty` component in a bunch of places so I'm going to use them for now and we can replace them all at once when we decide on a good icon, etc.

Is there a way to generate correlation data locally? Part of the reason I added the story was because I struggled to test some of this locally

Commit:
fix table header color

Commit:
add ids to tables

daibhin commented:
@thmsobrmlr I certainly fixed point 1) in [this commit](https://github.com/PostHog/posthog/pull/20852/commits/6b04c8eed063dfbe4821b4944ef7f969f62be0fb). And hopefully 2) too 🤞

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `webkit` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `webkit` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

thmsobrmlr commented:
> Is there a way to generate correlation data locally? Part of the reason I added the story was because I struggled to test some of this locally

Yeah it's definitely hard to create data/funnels that even have a correlation analysis. The easiest way is probably to use the "Activation" funnel from demo data.

thmsobrmlr commented:
> @thmsobrmlr I certainly fixed point 1) in [this commit](https://github.com/PostHog/posthog/pull/20852/commits/6b04c8eed063dfbe4821b4944ef7f969f62be0fb). And hopefully 2) too 🤞

Yup, (1) looks good now. (2) is still present, but I just found out the issue is also present in the version in `master`. Good to ship this then :)

Commit:
Merge branch 'master' into dn-chore/funnel-correlation-table
